spec/integration/check_runner_spec.rb

```diff
@@ -0,0 +1,167 @@
+require 'webmock/rspec'
+require 'checker'
+require 'fileutils'
+
+# integration test with in-memory db, webmocked services, empty whitelist, real csvs (removed afterwards)
+# TODO: suppress logging
+RSpec.describe CheckRunner do
+  before do
+    @publishing_api_content_and_links_url = "#{Plek.new.find('publishing-api')}/v2/grouped-content-and-links"
+    @publishing_api_lookup_url = "#{Plek.new.find('publishing-api')}/lookup-by-base-path"
+    @rummager_search_url = "#{Plek.new.find('rummager')}/unified_search.json"
+
+    @csvdir = tempdir
+
+    stub_publishing_api_content
+    stub_rummager_content
+  end
+
+  after do
+    cleanup
+  end
+
+  it "runs imports, runs checks, and generates output" do
+    runner = CheckRunner.new("CHECK_OUTPUT_DIR" => @csvdir, "SUPPRESS_PROGRESS" => "y")
+
+    exit_code = runner.run
+    expect(exit_code).to eq(1)
+
+    check_csvs_are_present
+
+    check_csv_content
+  end
+
+  def check_csvs_are_present
+    expected_csvs = [
+      'BasePathsMissingFromRummager.csv',
+      'BasePathsMissingFromPublishingApi.csv',
+      'LinkedBasePathsMissingFromPublishingApi.csv',
+      'LinksMissingFromRummager.csv',
+      'LinksMissingFromPublishingApi.csv',
+      'RummagerLinksNotIndexedInRummager.csv',
+      'ExpiredWhitelistEntries.csv',
+      'RummagerRedirectedLinks.csv',
+      'RummagerRedirects.csv',
+    ]
+    actual_csvs = Dir[File.join(@csvdir, '*')].map { |f| File.basename(f) }
+    expect(actual_csvs).to contain_exactly(*expected_csvs)
+  end
+
+  def check_csv_content
+    expect(read_csv('LinksMissingFromRummager.csv')).to eq("link_type,link_content_id,content_id,publishing_app,format\norganisations,42,1,app1,format1\n")
```

Comment:
Is it possible to stub out the filesystem in this test? Have it write to some [StringIO](http://ruby-doc.org/stdlib-2.3.0/libdoc/stringio/rdoc/StringIO.html) objects or something?